2009-02-08  Michael Tautschnig  <mt@debian.org>

	* lib/setup-storage/Parser.pm: Support :encrypt with all mountpoint
		specifications
Index: trunk/lib/setup-storage/Parser.pm
===================================================================
--- trunk.orig/lib/setup-storage/Parser.pm
+++ trunk/lib/setup-storage/Parser.pm	
@@ -510,27 +510,15 @@
           $FAI::partition_pointer = (\%FAI::configs)->{$FAI::device}->{volumes}->{$2};
         }
 
-    mountpoint: '-'
-        {
-          # this partition should not be mounted
-          $FAI::partition_pointer->{mountpoint} = "-";
-          $FAI::partition_pointer->{encrypt} = 0;
-        }
-        | 'swap'
-        {
-          # this partition is swap space, not mounted
-          $FAI::partition_pointer->{mountpoint} = "none";
-          $FAI::partition_pointer->{encrypt} = 0;
-        }
-        | m{^/\S*}
+    mountpoint: m{^(-|swap|/[^\s\:]*)(:encrypt)?}
         {
           # set the mount point, may include encryption-request
-          if ($item[ 1 ] =~ m{^(/[^:]*):encrypt$}) {
+          $FAI::partition_pointer->{mountpoint} = $1;
+          $FAI::partition_pointer->{mountpoint} = "none" if ($1 eq "swap");
+          if (defined($2)) {
             &FAI::in_path("cryptsetup") or die "cryptsetup not found in PATH\n";
-            $FAI::partition_pointer->{mountpoint} = $1;
             $FAI::partition_pointer->{encrypt} = 1;
           } else {
-            $FAI::partition_pointer->{mountpoint} = $item[ 1 ];
             $FAI::partition_pointer->{encrypt} = 0;
           }
         }
